qiskit_experiments/curve_analysis/__init__.py

```diff
@@ -143,5 +156,14 @@
     BlochTrajectoryAnalysis,
 )
 
+from .data_processing import (
```

Comment:
In particular it's `mean_xy_data` that's referenced in the `average_method` analysis option, the others may not be referenced. We should also consider renaming `data_processing.py` to avoid name confusion with the data processor module.

Reply:
Yes, I am not sure I fully understand what needs to be `curve_analysis/data_processing.py` versus `qiskit_experiments.data_processing` :slightly_smiling_face: It seems like only `multi_mean_xy_data`, `filter_data`, and `data_sort` are imported out of `data_processing.py`, so I wonder if we want these to be public, though we need to if we are referring to the docstring (maybe the discussion should be moved).